alibi_detect/base.py

```diff
@@ -119,16 +119,14 @@ def get_config(self) -> dict:  # TODO - move to BaseDetector once config save/lo
         if self.config is not None:
             # Get config (stored in top-level self)
             cfg = self.config
-            # Get low-level nested detector (if needed)
-            detector = self._detector if hasattr(self, '_detector') else self  # type: ignore[attr-defined]
-            detector = detector._detector if hasattr(detector, '_detector') else detector  # type: ignore[attr-defined]
             # Add large artefacts back to config
             for key in LARGE_ARTEFACTS:
-                if key in cfg:  # self.config is validated, therefore if a key is not in cfg, it isn't valid to insert
-                    cfg[key] = getattr(detector, key)
+                if key in cfg and hasattr(self._nested_detector, key):
```

Comment:
Just to clarify, the previous implementation would cause an opaque error if the key doesn't exist in the nested detector? What's a concrete example of a `cfg` for which this would happen?

Reply:
For the last implementation and the new one, we are working on the assumption that there is a one-to-one mapping between a detector's pydantic model and its args/kwargs, which are passed into `_set_config` as `inputs`. Therefore, in the previous implementation, if a `key` was in `inputs`, it was set to `None`. Then, later in `get_config`, we could safely check whether it needed to be set back to its actual value in `cfg` by checking if the `key` existed in `cfg` (because a `key` existing in `cfg` means it must have been in `inputs` earlier). 

The above is all good for the current set of `LARGE_ARTEFACTS = ['x_ref', 'c_ref', 'preprocess_fn']`, since we can safely  get the real `key` value back from its corresponding attribute in `_nested_detector` (note that the `_nested_detector` bit is because for detectors like `MMDDrift`, `x_ref` etc are stored in `self._detector.x_ref` not `self.x_ref`). The problem might arise in the future for some artefacts like `model`. For `ClassifierDrift` it would be fine because the `model` key is stored in the `self.model` attribute. For `RegressorUncertaintyDrift` we have a problem, because `model` isn't stored as an attribute anywhere. Another solution would be to store it, but the question then is where. It is a weird one because its `_nested_detector` is a `KSDrift` which itself doesn't have a `model` attribute:

https://github.com/SeldonIO/alibi-detect/blob/df7edca2f565b241062c5fdf4811ef50821e3ce6/alibi_detect/cd/model_uncertainty.py#L118-L127

So if we add a `model` attribute to `self` we're in a situation where `x_ref` would be in `self._detector.x_ref` but `model` is in `self.model`. 

The current implementation deals with this by simply skipping the `LARGE_ARTEFACT` procedure for the uncertainty detectors. I think this probably leads to duplication of the `model`, since it is in the `cfg` and the `self.model_fn` (in the future we could add a `model` `@property` to the uncertainty detectors to be more clever about this...). In any case, because `model` is more complciated, and I'm worried about state leaking into config (if the model has been trained), I've left `model` out of `LARGE_ARTEFACTS` for now.

**TL'DR**: The new implementation shouldn't actually change anything right now, but it should be more robust in any cases where an artefact is not stored in `_nested_detector`. `model` is one example, but there could be others I haven't thought of...